src/app/plan/components/Wizard/VolumesTable.tsx

```diff
@@ -16,316 +12,143 @@ import {
-                pvAction = rowVal.type;
-
-              }
-            }
-          }
-          return {
-            name: planVolume.name,
-            project: planVolume.pvc.namespace,
-            storageClass: planVolume.storageClass || 'None',
-            size: planVolume.capacity,
-            claim: planVolume.pvc.name,
-            type: pvAction,
-            details: '',
-            supportedActions: planVolume.supported.actions,
-          };
-        });
-      } else {
-        mappedPVs = discoveredPersistentVolumes.map(planVolume => {
-          const pvAction = 'copy'; // Default to copy
-          return {
-            name: planVolume.name,
-            project: planVolume.pvc.namespace,
-            storageClass: planVolume.storageClass || '',
-            size: planVolume.capacity,
-            claim: planVolume.pvc.name,
-            type: pvAction,
-            details: '',
-            supportedActions: planVolume.supported.actions,
-          };
-        });
-      }
-      setFieldValue('persistentVolumes', mappedPVs);
-      setRows(mappedPVs);
-    }
-  }, [currentPlan, isPollingStatus]); // Only re-run the effect if fetching value changes
-
+const VolumesTable: React.FunctionComponent<IVolumesTableProps> = ({
+  isFetchingPVResources,
+  pvResourceList,
+  persistentVolumes,
+  onTypeChange,
+}: IVolumesTableProps) => {
+  const columns = [
+    { title: 'PV name' },
+    { title: 'Claim' },
+    { title: 'Namespace' },
+    { title: 'Storage class' },
+    { title: 'Size' },
+    { title: 'Migration type' },
+    { title: 'Details' },
+  ];
 
-  //TODO: added this component level error state to handle the case of no PVs
-  // showing up after 3 checks of the interval. When the isPVError flag is checked,
-  // the volumes form will show this error. Need to add redux actions & state to encapsulate
-  // validation so that this error state enables the user to go to next page( that possibly
-  // shows a different set of forms catered to stateless apps)
-
-  if (isPVError) {
-    return (
-      <Grid gutter="md" className={styles.centerAlign}>
-        <GridItem>
-          <div className={styles.errorDiv}>
-            <StatusIcon isReady={false} />
-            PV Discovery Error
-          </div>
-        </GridItem>
-      </Grid>
-    );
-  }
-  if (isPollingStatus ||
-    (currentPlanStatus.state === 'Pending')) {
-    return (
-      <Bullseye>
-        <EmptyState variant="large">
-          <div className="pf-c-empty-state__icon">
-            <Spinner size="xl" />
-          </div>
-          <Title headingLevel="h2" size="xl">
-            Discovering persistent volumes attached to source projects...
-          </Title>
-        </EmptyState>
-      </Bullseye>
+  const rows = persistentVolumes.map((pv, pvIndex) => {
+    const matchingPVResource = pvResourceList.find(pvResource => pvResource.name === pv.name);
+    const migrationTypeOptions = pv.supportedActions.map(
```

Comment:
seeing the app blow up here in some cases. I think we need a guard if supportedActions is undefined. 

![image](https://user-images.githubusercontent.com/11218376/78188205-d397f600-743d-11ea-8f08-e2287ea61328.png)

Reply:
![image](https://user-images.githubusercontent.com/11218376/78188570-751f4780-743e-11ea-9e8a-0f3cbbb76294.png)

Reply:
Steps to reproduce: navigate back after selecting namespaces, select a new namespace & deselect old namespace, try to navigate forward and run pv discovery.

Reply:
Ah interesting... I wonder why you're getting the original PV from redux there and not the meta-PV-thingy defined here: https://github.com/konveyor/mig-ui/pull/776/files#diff-342db03f4d82174c47390c429f6aa190R77-R85 ... I'll take a look.

Reply:
It's strange that the same error wasn't being caused by this: https://github.com/konveyor/mig-ui/pull/776/files#diff-8d2e158fee7fbe8447115eb6c8a9725eL260-L264

Reply:
Wrote up an issue about this and the refactor I reference in a few of these comments: https://github.com/konveyor/mig-ui/issues/797